From 68a3347814c54defb8ed451f0e4be44e901e8498 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 15 May 2020 13:18:03 +0000
Subject: [PATCH] util.hpp, hub_service.hpp: include string to fix build with
 gcc-10

* fixes:
luna-service2/3.21.2-5-r23/git/include/private/util.hpp:107:91: error: return type 'std::string' {aka 'class std::__cxx11::basic_string<char>'} is incomplete
luna-service2/3.21.2-5-r23/git/include/private/util.hpp:109:39: error: invalid use of incomplete type 'const string' {aka 'const class std::__cxx11::basic_string<char>'}
luna-service2/3.21.2-5-r23/git/include/private/util.hpp:109:54: error: invalid use of incomplete type 'const string' {aka 'const class std::__cxx11::basic_string<char>'}
luna-service2/3.21.2-5-r23/git/src/ls-hubd/manifest.hpp:50:39: error: use of deleted function 'std::hash<std::__cxx11::basic_string<char> >::hash()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/basic_string.h:6798:12: error: redefinition of 'struct std::hash<std::__cxx11::basic_string<char> >'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/basic_string.h:6798:12: error: specialization of 'std::hash<std::__cxx11::basic_string<char> >' after instantiation
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/basic_string.h:6807:12: error: redefinition of 'struct std::__is_fast_hash<std::hash<std::__cxx11::basic_string<char> > >'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/basic_string.h:6807:12: error: specialization of 'std::__is_fast_hash<std::hash<std::__cxx11::basic_string<char> > >' after instantiation
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/functional_hash.h:101:12: error: no matching function for call to 'std::__hash_enum<std::__cxx11::basic_string<char>, false>::__hash_enum()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/functional_hash.h:101:12: error: 'std::__hash_enum<_Tp, <anonymous> >::~__hash_enum() [with _Tp = std::__cxx11::basic_string<char>; bool <anonymous> = false]' is private within this context
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/functional_hash.h:101:12: error: use of deleted function 'std::__hash_enum<std::__cxx11::basic_string<char>, false>& std::__hash_enum<std::__cxx11::basic_string<char>, false>::operator=(const std::__hash_enum<std::__cxx11::basic_string<char>, false>&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable.h:1066:35: error: use of deleted function 'std::__detail::_Hashtable_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<true, false, true> >& std::__detail::_Hashtable_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<true, false, true> >::operator=(const std::__detail::_Hashtable_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<true, false, true> >&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable.h:1389:5: error: use of deleted function 'std::__detail::_Hashtable_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<true, false, true> >::~_Hashtable_base()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable.h:1389:5: error: use of deleted function 'std::__detail::_Hashtable_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> (HubService::*)(LSTransportMessage*, const char*)>, std::__detail::_Select1st, std::equal_to<std::__cxx11::basic_string<char> >, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Hashtable_traits<true, false, true> >::~_Hashtable_base()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1109:12: error: use of deleted function 'std::hash<std::__cxx11::basic_string<char> >::~hash()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1109:12: error: use of deleted function 'std::hash<std::__cxx11::basic_string<char> >& std::hash<std::__cxx11::basic_string<char> >::operator=(const std::hash<std::__cxx11::basic_string<char> >&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1341:12: error: use of deleted function 'std::__detail::_Hashtable_ebo_helper<1, std::hash<std::__cxx11::basic_string<char> >, true>::~_Hashtable_ebo_helper()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1341:12: error: use of deleted function 'std::__detail::_Hashtable_ebo_helper<1, std::hash<std::__cxx11::basic_string<char> >, true>& std::__detail::_Hashtable_ebo_helper<1, std::hash<std::__cxx11::basic_string<char> >, true>::operator=(const std::__detail::_Hashtable_ebo_helper<1, std::hash<std::__cxx11::basic_string<char> >, true>&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1377:16: error: static assertion failed: hash function must be invocable with an argument of key type
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1379:16: error: no match for call to '(const std::hash<std::__cxx11::basic_string<char> >) (const std::__cxx11::basic_string<char>&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1725:10: error: use of deleted function 'std::__detail::_Hash_code_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, true>::~_Hash_code_base()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1725:10: error: use of deleted function 'std::__detail::_Hash_code_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, true>& std::__detail::_Hash_code_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, true>::operator=(const std::__detail::_Hash_code_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, Groups>, std::__detail::_Select1st, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, true>&)'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/hashtable_policy.h:1725:10: error: use of deleted function 'std::__detail::_Hash_code_base<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> (HubService::*)(LSTransportMessage*, const char*)>, std::__detail::_Select1st, std::hash<std::__cxx11::basic_string<char> >, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, true>::~_Hash_code_base()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/stl_pair.h:217:11: error: 'std::pair<_T1, _T2>::first' has incomplete type
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/bits/unordered_map.h:228:28: error: use of deleted function 'std::hash<std::__cxx11::basic_string<char> >::hash()'
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/type_traits:1093:52: error: static assertion failed: template argument must be a complete class or an unbounded array
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/type_traits:1114:52: error: static assertion failed: template argument must be a complete class or an unbounded array
luna-service2/3.21.2-5-r23/recipe-sysroot/usr/include/c++/10.1.0/utility:179:32: error: body of 'constexpr' function 'static constexpr _Tp1& std::__pair_get<0>::__get(std::pair<_T1, _T2>&) [with _Tp1 = const std::__cxx11::basic_string<char>; _Tp2 = std::__cxx11::basic_string<char> (HubService::*)(LSTransportMessage*, const char*)]' not a return-statement
---
 include/private/util.hpp    | 1 +
 src/ls-hubd/hub_service.hpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/private/util.hpp b/include/private/util.hpp
index 53513f7..3ee165d 100644
--- a/include/private/util.hpp
+++ b/include/private/util.hpp
@@ -19,6 +19,7 @@
 
 #include <memory>
 #include <utility>
+#include <string>
 
 #include <glib.h>
 
diff --git a/src/ls-hubd/hub_service.hpp b/src/ls-hubd/hub_service.hpp
index a97a0c8..f7dde51 100644
--- a/src/ls-hubd/hub_service.hpp
+++ b/src/ls-hubd/hub_service.hpp
@@ -17,6 +17,7 @@
 #ifndef _HUB_SERVICE_HPP_
 #define _HUB_SERVICE_HPP_
 
+#include <string>
 #include <set>
 #include <memory>
 #include <unordered_map>
